### PR TITLE
Add DDP multi-GPU training with AMP and gradient accumulation

### PR DESCRIPTION
**Pull request by sreevadde**
## Summary

This PR adds comprehensive multi-GPU training support using DistributedDataParallel (DDP):

- **DDP Training**: Efficient multi-GPU training via `torchrun`
- **AMP (Automatic Mixed Precision)**: ~2x faster training with `--amp` flag
- **Gradient Accumulation**: Larger effective batch sizes with `--accum-steps N`
- **Distributed Validation**: Periodic evaluation during training with `--eval-freq N`
- **Multi-node Support**: Compatible with `torchrun` for multi-node training
- **Linear LR Scaling**: Automatic learning rate adjustment by `num_gpus × accum_steps`

## Usage

```bash
# Basic 4-GPU training
torchrun --nproc_per_node=4 train_ddp.py configs/thumos_i3d.yaml --output exp

# With AMP (faster training)
torchrun --nproc_per_node=4 train_ddp.py configs/thumos_i3d.yaml --amp --output exp

# With gradient accumulation
torchrun --nproc_per_node=4 train_ddp.py configs/thumos_i3d.yaml --accum-steps 2 --output exp

# Full example with validation
torchrun --nproc_per_node=4 train_ddp.py configs/thumos_i3d.yaml \
    --amp --accum-steps 2 --eval-freq 5 --output exp
```

## Implementation Details

- Self-contained `train_ddp.py` with all DDP logic (no changes to core libs)
- Gradient sync optimization with `model.no_sync()` during accumulation steps
- Proper checkpoint saving with AMP scaler state
- Rank-aware logging (only main process prints)
- Results gathered from all processes for distributed validation

## Changes

- `train_ddp.py`: New DDP training script with AMP and gradient accumulation
- `README.md`: Updated documentation with DDP usage examples
- No changes to existing single-GPU training workflow

### CODE REVIEW

**Comment by sreevadde** [comment]
Keeping as standalone fork improvement
